app/src/main/java/com/khinthirisoe/bakingapp/ui/baking/view/BakingActivity.kt

```diff
@@ -0,0 +1,77 @@
+package com.khinthirisoe.bakingapp.ui.baking.view
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.widget.ProgressBar
+import android.widget.Toast
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.khinthirisoe.bakingapp.R
+import com.khinthirisoe.bakingapp.data.model.Recipe
+import com.khinthirisoe.bakingapp.di.component.AppComponent
+import com.khinthirisoe.bakingapp.di.component.DaggerActivityComponent
+import com.khinthirisoe.bakingapp.di.module.ActivityModule
+import com.khinthirisoe.bakingapp.ui.baking.BakingContract
+import com.khinthirisoe.bakingapp.ui.base.BaseActivity
+import com.khinthirisoe.bakingapp.ui.ingredients.IngredientsActivity
+import javax.inject.Inject
+
+
+class BakingActivity : BaseActivity(), BakingContract.View, BakingAdapter.BakingRecyclerViewClickListener {
+
+    @Inject
+    lateinit var presenter: BakingContract.Presenter
+
+    private lateinit var bakingRecyclerView: RecyclerView
+    private lateinit var progressBar: ProgressBar
+    private var bakingAdapter: BakingAdapter? = null
+
+    override fun setupComponent(appComponent: AppComponent) {
+        val component =
+            DaggerActivityComponent.builder().appComponent(appComponent).activityModule(ActivityModule(this)).build()
+        component.inject(this)
+
+        presenter.onAttachView(this)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(com.khinthirisoe.bakingapp.R.layout.activity_baking)
+
+        initView()
+
+        presenter.fetchBakingRecipes()
+    }
+
+    private fun initView() {
+
+        bakingRecyclerView = findViewById(R.id.recyclerView)
+        progressBar = findViewById(R.id.progressBar)
+
+        val mLayoutManager = LinearLayoutManager(this)
+        bakingRecyclerView.layoutManager = mLayoutManager
+
+    }
+
+    override fun showBakingLists(list: ArrayList<Recipe>) {
+        bakingAdapter = BakingAdapter(this, list as MutableList<Recipe>, this)
```

Comment:
@KhinThiriSoe do not instantiate the adapter every time you get the data. Just instantiate the adapter in **initView** and use something like ** bakingAdapter.setRecipes(recipes)** here

Reply:
by instantiating every time you get the data, it consumes more memory